api/proxy/store/generated_key/v2/eigenda.go

```diff
@@ -264,8 +269,11 @@ func (e Store) VerifyCert(ctx context.Context, versionedCert certs.VersionedCert
 		return err
 	}
 
+	timeoutCtx, cancel := context.WithTimeout(ctx, e.contractCallTimeout)
+	defer cancel()
+
 	// verify cert via simulation call to verifier contract
-	err = e.certVerifier.CheckDACert(ctx, sumDACert)
+	err = e.certVerifier.CheckDACert(timeoutCtx, sumDACert)
```

Comment:
interesting. Don't really like passing a `contractCallTimeout` to checkDACert here because checkDACert is actually making 2 eth Calls, so the 5sec default timeout here is actually used for the entirety of CheckDACert whereas it should probably be used on a per-call basis. In either case given that the payload disperser is doing the exact same thing its prob fine to merge.

I think an ideal solution would prob have the network call timeouts on the eth-client itself instead of the higher-level structs, but that would be a much larger refactor.